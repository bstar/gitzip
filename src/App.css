*:focus {outline:none}

.App {
  text-align: center;
}

input[type=text] {
  border-radius: 10px;
}

.App-logo {
  padding: 30px;
  height: 30vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.main-container {
  background-color:rgb(42, 53, 88);
  min-height: 100vh;
  font-size: calc(10px + 2vmin);
}

.content-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;

}

@media only screen and (max-width: 600px) {
  .repos-container .list {
    max-height: 400px;
    overflow: scroll;
    border-top: 1px solid rgb(145, 167, 226);
    border-bottom: 1px solid rgb(145, 167, 226);
    border-left: 1px solid rgb(61, 69, 94);
    border-right: 1px solid rgb(61, 69, 94);
    box-shadow: 0px 0px 10px rgba(24, 20, 20, .3), 0px -10px 5px rgba(24, 20, 20, .2);
  }
}

.header {
  color: #eee;
  font-size: 14px;
  display: flex;
  padding: 10px;
}

.header img {
  width: 40px;
  border-radius: 50%;
}

.header div {
  padding: 10px 0px 0px 10px;
}

.user-text {
}

.App {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.auth-input {
  outline: none;
  padding: 15px;
  border: 1px solid rgb(252, 255, 69);
  font-size: 16px;
  color: #eee;
  background-color: rgb(42, 53, 88);
  width: 450px;
  text-align: center;
}

.content-header {
  text-shadow: 1px 1px 2px black;
  padding: 20px 0px 20px 10px;
  margin-top: 15px;
  color: #eee;
}

.repo-card-container {
  display: block;
  text-align: left;
  color: #eee;
  font-size: 14px;
  background-color: rgba(38, 47, 72, 0.747);
  border: 1px solid rgb(87, 104, 119);
  padding: 10px;
  margin: 10px;
  width: 300px;
  box-shadow: 0px 2px 14px -5px rgba(0,0,0,0.75);
}

.repo-card-container div {
  padding: 2px;
}

.repo-card-container:hover {
  background-color: rgb(49, 61, 92);
  cursor: pointer;
}

.issue-card-container {
  display: block;
  text-align: left;
  color: #eee;
  font-size: 14px;
  background-color: rgba(38, 47, 72, 0.747);
  border: 1px solid rgb(87, 104, 119);
  padding: 10px;
  margin: 10px;
  width: 300px;
  border-radius: 15px 0px 15px 0px;
  box-shadow: 0px 2px 14px -5px rgba(0,0,0,0.75);
}

.issue-card-container div {
  padding: 2px;
}

.issue-card-container:hover {
  background-color: rgb(49, 61, 92);
  cursor: pointer;
}

.warning {
  color: rgb(245, 153, 99);
  padding: 10px;
  font-size: 18px;
}