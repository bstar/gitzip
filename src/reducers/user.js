import {
    SET_GITHUB_KEY,
    GET_USER_SUCCESS,
    GET_USER_REPOS_SUCCESS,
    SET_ACTIVE_REPO,
    GET_USER_REPO_ISSUES_SUCCESS,
} from '../constants/ActionTypes';

const initialSettings = {};

const user = (state = initialSettings, action) => {

    switch (action.type) {

        case SET_GITHUB_KEY:

            return {
                ...state,
                key: action.payload,
            };

        case GET_USER_SUCCESS:

            return {
                ...state,
                login: action.payload.login,
                avatar_url: action.payload.avatar_url,
                url: action.payload.url,
                name: action.payload.name,
            };

        case GET_USER_REPOS_SUCCESS:

            return {
                ...state,
                repos: action.payload,
            };

        case SET_ACTIVE_REPO:

            return {
                ...state,
                activeRepoData: action.payload,
                issues: null,
            };

        case GET_USER_REPO_ISSUES_SUCCESS:

            return {
                ...state,
                issues: action.payload,
            };

        default:
            return state;
    }
};

export default user;
