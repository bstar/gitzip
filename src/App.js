import React, { Component } from 'react';
import { connect } from 'react-redux';
import logo from './Octocat.png';
import Auth from './components/auth';
import Repos from './components/repos';
import Issues from './components/issues';
import './App.css';


const mapStateToProps = state => {

  return ({
    user: state.user,
    issues: state.user.issues,
  });
};

class App extends Component {

  render () {

    const { user, issues } = this.props;
    const { login, repos } = user;

    return (
      <div className="main-container">
        { user.login &&
          <div className="header">
            <img alt="avatar" src={user.avatar_url} />
            <div className="user-text">{user.name} ({user.login})</div>
          </div>
        }
        { login ?
          <div className="content-container">

            <Repos repos={repos} />
  
            { issues &&
                <Issues issues={issues} user={user} />
            }

          </div>
        :
          <div className="App">
            <Auth />
            <img src={logo} className="App-logo" alt="logo" />
          </div>
        }
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
)(App);
