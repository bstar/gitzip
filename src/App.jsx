import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import logo from './Octocat.png';
import Auth from './components/auth';
import Repos from './components/repos';
import Issues from './components/issues';
import './App.css';


const mapStateToProps = state => ({
    user: state.user,
    issues: state.user.issues,
});

const App = ({ user, issues }) => {

    const { login, repos } = user;

    return (
        <div className="main-container">
            { login ?
                <div>
                    <div className="header">
                        <img alt="avatar" src={user.avatar_url} />
                        <div className="user-text">{user.name} ({user.login})</div>
                    </div>
                    <div className="content-container">
                        <Repos repos={repos} />
                        { issues &&
                            <Issues issues={issues} user={user} />
                        }
                    </div>
                </div> :
                <div className="App">
                    <Auth />
                    <img src={logo} className="App-logo" alt="logo" />
                </div>
            }
        </div>
    );
};

App.propTypes = {
    user: PropTypes.object,
    issues: PropTypes.array,
};

export default connect(
    mapStateToProps,
)(App);
