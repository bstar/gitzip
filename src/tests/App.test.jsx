import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../reducers';
import App from '../App';

const store = createStore(
    reducers,
);

it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><App /></Provider>).toJSON();

    expect(tree).toMatchSnapshot();
});
