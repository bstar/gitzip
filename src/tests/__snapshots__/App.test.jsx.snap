// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="App"
  >
    <input
      className="auth-input"
      name="key"
      onKeyPress={[Function]}
      placeholder="Format: username:key"
      type="text"
    />
    <img
      alt="logo"
      className="App-logo"
      src="Octocat.png"
    />
  </div>
</div>
`;
